### PR TITLE
test(clerk-js): Test appearance

### PR DESCRIPTION
**Pull request by desiprisg**
## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [x] other: tests

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [ ] `@clerk/shared`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

### CODE REVIEW

File: packages/clerk-js/src/ui/components/ImpersonationFab/ImpersonationFab.tsx [hunk 1]
```diff
@@ -146,7 +146,7 @@ const _ImpersonationFab = () => {
       window.innerWidth - current.offsetLeft - current.offsetWidth - e.movementX
     }px`;
     document.documentElement.style.setProperty(rightProperty, rightOffestBasedOnViewportAndContent);
-    document.documentElement.style.setProperty(topProperty, `${current.offsetTop + e.movementY}px`);
+    document.documentElement.style.setProperty(topProperty, `${current.offsetTop - -e.movementY}px`);
```
Comment (anagstef) [inline packages/clerk-js/src/ui/components/ImpersonationFab/ImpersonationFab.tsx:149]: ❓Is this for tsc errors?

File: packages/clerk-js/src/ui/customizables/__tests__/elementDescriptors.test.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+          data-testid='test'
+          elementDescriptor={descriptors.socialButtonsIconButton}
+        />
+      </AppearanceProvider>,
+    );
+
+    expect(screen.getByTestId('test')).toHaveClass('cl-socialButtonsIconButton');
+  });
+
+  it('id related stable classname is added to the element', () => {
```
Comment (anagstef) [inline packages/clerk-js/src/ui/customizables/__tests__/elementDescriptors.test.tsx:23]: 🙃 We can test all the social providers with `it.each`

File: packages/clerk-js/src/ui/customizables/__tests__/makeCustomizable.test.tsx [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+import { render, screen } from '../../../testUtils';
+import React from 'react';
+
+import { Box, descriptors } from '..';
+import { AppearanceProvider } from '../AppearanceContext';
+import { knownColors } from './testUtils';
+import { InternalThemeProvider } from '../../styledSystem';
+
+describe('Theme used in sx callback', () => {
+  it('styles match the theme/globalAppearance', () => {
```
Comment (anagstef) [inline packages/clerk-js/src/ui/customizables/__tests__/makeCustomizable.test.tsx:11]: 🙃 Also `it.each` if we want more tests with different combinations

**Review by anagstef — APPROVED** [review]
Excellent! (Note: is this PR adding an empty file with path `packages/react/src/contexts/__tests__/ClerkProvider.test.tsx` ?)

**Comment by nikosdouvlis** [comment]
Hey @desiprisg , are we going to create a separate PR for localization tests?

**Comment by desiprisg** [comment]
Yes, we'll do it in a seperate PR

File: packages/clerk-js/src/ui/components/ImpersonationFab/ImpersonationFab.tsx (hunk 1 above)
Comment (desiprisg) [reply packages/clerk-js/src/ui/components/ImpersonationFab/ImpersonationFab.tsx:149]: It's to satisfy the linter for now

File: packages/clerk-js/src/ui/customizables/__tests__/elementDescriptors.test.tsx (hunk 2 above)
Comment (desiprisg) [reply packages/clerk-js/src/ui/customizables/__tests__/elementDescriptors.test.tsx:22]: Not sure if it matters here, it's not what this is supposed to be testing. The socialButtons was just a good example that has everything!

**Review by SokratisVidros — APPROVED** [review]

**Pull request merged**